### ISSUE TITLE
pylint fails to mix noneditable and editable packages sharing a pkg_resources-style namespace

### ISSUE THREAD

**Issue by p4l1ly** [pylint-dev/pylint#8896]
### Bug description

When a noneditable installed project contains a pkg_resources-style or pkgutil-style namespace package `top` (containing any `__init__.py`) and an editable installed project contains a python module `top.x`, pylint reports import errors for `top.x` but python itself is ok to import it. If using new style native namespace packages (containing no `__init__.py`) or if both projects are installed as editable or both are installed as noneditable, everything is ok - pylint reports no errors.
See the branches `native, pkgutil, pkg_resources` in this repo: https://github.com/p4l1ly/pylint_editable_test

Supporting old style namespacing techinques might seem obsolete but even the official Python documentation states that they are still used in most of the existing namespace packages.

**EDIT:** After reading [this piece of documentation](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/) about namespace packages, I've added some standard terminology.

### Configuration

default

### Command used

```shell
git clone https://github.com/p4l1ly/pylint_editable_test
cd pylint_editable_test
python3 -m venv venv
. ./venv/bin/activate
pip install 'pylint==3.0.0a6'

git checkout pkgutil  # or pkg_resources or native
# removing '-e' or adding '-e' to both dependencies results in no pylint errors
pip install ./noneditable_dep -e ./editable_dep
rm -r */build */*.egg-info
python3 test.py  # no errors
python3 -m pylint test.py  # errors
```

### Pylint output

```shell
************* Module test
test.py:3:0: E0401: Unable to import 'top.editable_dep' (import-error)
test.py:3:0: E0611: No name 'editable_dep' in module 'top' (no-name-in-module)
test.py:5:0: E1101: Module 'top' has no 'editable_dep' member (no-member)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 3.0.0a6  # tried as well with 2.17.5
astroid 2.15.6
Python 3.11.3 (main, Jun  5 2023, 09:32:32) [GCC 13.1.1 20230429]
```

### OS / Environment

Arch Linux 6.4.6 kernel, x86_64,

**Comment by DanielNoord** [comment]
Closing as duplicate of https://github.com/pylint-dev/pylint/issues/8829.
